gtk/src/functions.rs

```diff
@@ -0,0 +1,43 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::translate::*;
+use std::ptr;
+
+#[doc(alias = "gtk_accelerator_parse_with_keycode")]
+pub fn accelerator_parse_with_keycode(
+    accelerator: &str,
+) -> Option<(u32, Vec<u32>, gdk::ModifierType)> {
+    assert_initialized_main_thread!();
+    unsafe {
+        let mut accelerator_key = std::mem::MaybeUninit::uninit();
+        let mut accelerator_codes_ptr = ptr::null_mut();
+        let mut accelerator_mods = std::mem::MaybeUninit::uninit();
+        ffi::gtk_accelerator_parse_with_keycode(
+            accelerator.to_glib_none().0,
+            accelerator_key.as_mut_ptr(),
+            &mut accelerator_codes_ptr,
+            accelerator_mods.as_mut_ptr(),
+        );
+        if !accelerator_codes_ptr.is_null() {
+            let mut accelerator_codes = Vec::new();
+            let mut i = 0;
+            loop {
+                // zero-terminated array
+                let ret = std::ptr::read(accelerator_codes_ptr.add(i));
+                if ret == 0 {
+                    break;
+                }
+                accelerator_codes.push(ret);
+                i += 1;
+            }
+            glib::ffi::g_free(accelerator_codes_ptr as *mut _);
```

Comment:
If you feel like it

```rust
let mut len = 0;
if !accelerator_codes_ptr.is_null() {
    while ptr::read(accelerator_codes_ptr.add(len)) != 0 {
       len += 1;
    }
}
let accelerator_codes = glib::GSlice::from_glib_full_num(accelerator_codes_ptr, len);
```

That way you don't have to unnecessarily allocate twice but can re-use the allocation from GTK here.

Reply:
:o

Reply:
Well, that's why I started adding `glib::GSlice` and things like that. I have some WIP changes locally for more things and having gir autogenerate that (in other cases, not here, this function is complicated).

Reply:
Done!